extract/extract.py

```diff
@@ -256,6 +244,27 @@ def execute(self, request: ServiceRequest):
             extracted, password_protected = self.extract_zip(request, request.file_path, request.file_type)
             summary_section_heuristic = 19
 
+        if request.file_type in ["archive/gzip", "archive/zstd"] and not password_protected and len(extracted) == 1:
```

Comment:
FYI: As I understand the code, this case is a special handling for `.tar.*` files, but I think it's too restricted: just the [Python standard](https://docs.python.org/3/library/tarfile.html?highlight=tarfile#module-tarfile) library supports gzip, bzip2 and lzma. I'd suggest either extending the list for all cases from the standard library, or even make it general for _every_ archive containing just the tarfile.

Reply:
Just tested with a bzip2, and we're handling it without jumping over the middle tar. Definitely a nice optimization. I'll widen the second stage tar extraction to all files types handled in Extract. Thank you for the review!